backend/app/services/users/sign_in.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Users
+  class SignIn < Core::Service
+    def call(email:, password:)
+      @email = email
+      @password = password
+      @user = yield find_user(@email)
+      validation = validate_password(@password)
+      # binding.pry
+      if validation.instance_of?(Dry::Monads::Result::Success)
+        token, refresh_token = yield issue_token(@user)
+        Success[token, refresh_token]
+      else
+        validation
+      end
```

Comment:
You could handle the same flow using something like this:

```ruby
yield validate_password(@password)
token, refresh_token = yield issue_token(@user)
Success[token, refresh_token]
```

`yield validate_password(@password)` will return `Failure(:wrong_password)` in case of failure and will not trigger other lines below.